dev/Saumya/Train-test split traversal.ipynb

```diff
@@ -0,0 +1,1638 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us now seperate the dependent feature and independent features. Here, Class is the dependent feature."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X = df.drop(labels='Class', axis=1)  # Drop the target attribute from the X variable\n",
+    "y = df.Class  # The target attribute to be predicted"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Classifier and metric selection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we create a Data Frame which will be used to store test ratios and performance scores."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "metric_cols = [\n",
+    "    \"Test Ratio\",\n",
+    "    \"Train Score\",\n",
+    "    \"Test Score\",\n",
+    "    \"Recall Score Macro\",\n",
+    "    \"f1 Score Macro\"\n",
+    "]\n",
+    "\n",
+    "metric_table = pd.DataFrame(columns=metric_cols)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We have used the following three classifiers:\n",
+    "1. Random forest Classifier- It uses a number of decision tree classifiers\n",
+    "2. Support Vector Classification- Its implementation is based on libsvm\n",
+    "3. Quadratic Discriminant Analysis- It works by fitting class conditional densities\n",
+    "\n",
+    "We have chosen these to study the effect on different **types** of classifiers."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "MLA = [\n",
+    "        ensemble.RandomForestClassifier(),\n",
+    "\n",
+    "        svm.SVC(),\n",
+    "    \n",
+    "        discriminant_analysis.QuadraticDiscriminantAnalysis(),\n",
+    "        ]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tables and graphs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This function creates a table with the given range of test ratio for a given classifier. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_scores(start, end, interval, clf):\n",
+    "    result_list = []\n",
+    "    for i in range (start, end, interval):\n",
+    "        result_dict = {}\n",
+    "        test_ratio = i/100\n",
+    "        \n",
+    "        result_dict[\"Test Ratio\"] = test_ratio\n",
+    "        \n",
+    "        train_X, test_X, train_y, test_y = model_selection.train_test_split(X, y, test_size=test_ratio, stratify=y, random_state=42)\n",
+    "        train_X = (train_X - train_X.mean())/train_X.std()\n",
+    "        test_X = (test_X - test_X.mean())/test_X.std()\n",
```

Comment:
You should be normalizing the test set using the mean and std of the __training__ set, same as for the training set itself. Thinking of the normalization parameters as part of the model, this ensures that your model is not learning anything from the test set. In fact you can use the built-in [StandardScaler](https://scikit-learn.org/stable/modules/preprocessing.html#standardization-or-mean-removal-and-variance-scaling) to handle this.